plugins/catalog-import/src/components/useImportState.ts

```diff
@@ -0,0 +1,262 @@
+/*
+ * Copyright 2021 Spotify AB
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
```

Comment:
as we talked about, this is a bit meaty - could perhaps be written in more of a reducer-style, but i don't have a very strong opinion

Reply:
I refactored it to `useReducer`. Though the code is not necessarily shorter, I hope it is a cleaner.